pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -338,7 +338,76 @@ func routesForGrant(l logr.Logger, client kube_client.Client) kube_handler.MapFu
 	}
 }
 
+// routesForService returns a function that calculates which HTTPRoutes might
+// be affected by changes in a Service.
+func routesForService(l logr.Logger, client kube_client.Client) kube_handler.MapFunc {
+	l = l.WithName("service-to-routes-mapper")
+
+	return func(ctx context.Context, obj kube_client.Object) []kube_reconcile.Request {
+		svc, ok := obj.(*kube_core.Service)
+		if !ok {
+			l.Error(nil, "unexpected error converting object to Service", "typ", reflect.TypeOf(obj))
+			return nil
+		}
+
+		var routes gatewayapi.HTTPRouteList
+		if err := client.List(ctx, &routes, kube_client.MatchingFields{
+			servicesOfRouteField: kube_client.ObjectKeyFromObject(svc).String(),
+		}); err != nil {
+			l.Error(nil, "unexpected error listing HTTPRoutes")
```

Comment:
Don't we want to log an underlying error here?
```suggestion
			l.Error(err, "unexpected error listing HTTPRoutes")
```